commit 2c9bf9f58b6237942960d2cd4f1fc91a2e6beb37
Author: Jose Dapena Paz <jdapena@igalia.com>
Date:   Tue Mar 1 18:06:38 2022 +0000

    GCC: explicitely move return value of SCTHashdanceMetadata::ToValue
    
    GCC rejects to do Return Value Optimization in
    SCTHashdanceMetadata::ToValue, because the copy constructor is
    deleted, and in that scenario RVO is rejected in GCC:
        ../../services/network/sct_auditing/sct_auditing_reporter.cc: In member function ‘base::Value network::SCTAuditingReporter::SCTHashdanceMetadata::ToValue() const’:
        ../../services/network/sct_auditing/sct_auditing_reporter.cc:191:10: error: use of deleted function ‘base::Value::Value(const base::Value&)’
          191 |   return value;
              |          ^~~~~
        In file included from ../../services/network/sct_auditing/sct_auditing_reporter.h:14,
                         from ../../services/network/sct_auditing/sct_auditing_reporter.cc:5:
        ../../base/values.h:254:3: note: declared here
          254 |   Value(const Value&) = delete;
              |   ^~~~~
    
    Bug: 819294
    Change-Id: I111e51dd10eee7b909d4ac3c0911aac18a589166
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3488747
    Reviewed-by: Chris Thompson <cthomp@chromium.org>
    Commit-Queue: José Dapena Paz <jdapena@igalia.com>
    Cr-Commit-Position: refs/heads/main@{#976299}

diff --git a/services/network/sct_auditing/sct_auditing_reporter.cc b/services/network/sct_auditing/sct_auditing_reporter.cc
index 900a05dc81b0b..bec982d487428 100644
--- a/services/network/sct_auditing/sct_auditing_reporter.cc
+++ b/services/network/sct_auditing/sct_auditing_reporter.cc
@@ -194,7 +194,7 @@ base::Value SCTAuditingReporter::SCTHashdanceMetadata::ToValue() const {
       kLogIdKey, base::Base64Encode(base::as_bytes(base::make_span(log_id))));
   value.SetKey(kLogMMDKey, base::TimeDeltaToValue(log_mmd));
   value.SetKey(kCertificateExpiry, base::TimeToValue(certificate_expiry));
-  return value;
+  return std::move(value);
 }
 
 // static
